packages/forms-web-app/src/controllers/document-service-proxy.js

```diff
@@ -0,0 +1,34 @@
+const { getDocument } = require('../lib/documents-api-wrapper');
+const logger = require('../lib/logger');
+
+exports.getDocument = async (req, res) => {
+  const { appealId, documentId } = req.params;
+
+  if (!appealId || !documentId) {
+    logger.error('Request was missing either an appealId, or a documentId');
+    return res.sendStatus(400);
+  }
+
+  let fileResponse;
+  try {
+    fileResponse = await getDocument(appealId, documentId);
+  } catch (e) {
+    logger.error({ e }, 'Document service proxy lookup failed.');
+    return res.sendStatus(400);
+  }
+
+  try {
+    const fileName = fileResponse.headers.get('x-original-file-name');
+
+    res.set({
+      'content-length': fileResponse.headers.get('content-length'),
+      'content-disposition': `attachment;filename="${fileName}"`,
+      'content-type': fileResponse.headers.get('content-type'),
+    });
+  } catch (e) {
+    logger.error({ e }, 'Document was retrieved but appears to be missing key data.');
+    return res.status(500);
+  }
+
+  return fileResponse.body.pipe(res);
+};
```

Comment:
This endpoint is deliberately externally vague about what went wrong - just the `400` / `500` response code. 

Internally the logs should be useful in identifying which step failed. If something does go wrong, there are very few reasons this could have happened.

Either the document service API was down, or the appeal doesn't exist (very unlikely) or the document doesn't exist (also highly unlikely). 

One possibility is that the filename is missing - this *should* be highly unlikely going forwards, however it *may* fail on older records. This was the purpose of updating the DB with Paul's script to ensure all the expected fields are on every file upload. 

Finally, the downloaded file should be passed on / piped to the client, so this (afaik) should use as little memory as possible when working with large files <- sanity check please

Reply:
:+1: